CRM/Contact/BAO/Query.php

```diff
@@ -6363,7 +6368,6 @@ protected function prepareOrderBy($sort, $sortByChar, $sortOrder, $additionalFro
           // is not declared for them.
           // @todo so far only integer fields are being handled. If we add string fields we need to look at
           // escaping.
-          $fieldSpec = $this->getMetadataForRealField($field);
           $pseudoConstantMetadata = CRM_Utils_Array::value('pseudoconstant', $fieldSpec, FALSE);
```

Comment:
Test failure seems to be due to the above line removal. I can replicate the same on search builder after submitting `state = not empty` condition -

![image](https://user-images.githubusercontent.com/5929648/54602102-b7116180-4a66-11e9-954b-74cc41cbc856.png)

Similar notice error is displayed at the bottom of https://test.civicrm.org/job/CiviCRM-Core-PR/25244/console.

    not ok 397 - Failure: api_v3_ContributionPageTest::testSubmitMembershipBlockIsSeparatePayment
      ---
      message: 'Failure in api call for contribution_page submit:  Undefined variable: fieldSpec'
      severity: fail
      data:
          got: 1
          expected: 0
Maybe, `$fieldSpec` assignment in line 6338 should be out of the if condition?

Reply:
oh yeah - I moved it further up but then it would up in the if

Reply:
OK - I moved it back out of the IF - guess we'll see if it causes any enotices now